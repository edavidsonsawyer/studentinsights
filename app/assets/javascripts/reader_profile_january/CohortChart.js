import React from 'react';
import PropTypes from 'prop-types';
import qs from 'query-string';
import GenericLoader from '../components/GenericLoader';
import {apiFetchJson} from '../helpers/apiFetchJson';
import BoxChart from './BoxChart';


export default class CohortChart extends React.Component {
  render() {
    const {gradeNow, readerJson, benchmarkAssessmentKey} = this.props;
    const url = `/reading/cohort?${qs.stringify({grade: gradeNow})}`;
    return (
      <div className="CohortChart">
        <GenericLoader
          promiseFn={() => apiFetchJson(url)}
          render={json => (
            <BoxChart
              gradeNow={gradeNow}
              readerJson={readerJson}
              benchmarkAssessmentKey={benchmarkAssessmentKey}
              renderCellFn={({dataPoint}) => {
                const value = dataPoint ? dataPoint.json.value : null;
                return value;
              }}
            />
          )}
        />
      </div>
    );
  }
}
CohortChart.propTypes = {
  gradeNow: PropTypes.string.isRequired,
  benchmarkAssessmentKey: PropTypes.string.isRequired,
  readerJson: PropTypes.object.isRequired
};
