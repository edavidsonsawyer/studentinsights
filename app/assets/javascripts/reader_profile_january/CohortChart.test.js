import React from 'react';
import ReactDOM from 'react-dom';
import {withNowContext} from '../testing/NowContainer';
import {expectSnapshotToMatchAfterTick} from '../testing/snapshotAsync';
import fetchMock from 'fetch-mock/es5/client';
import PerDistrictContainer from '../components/PerDistrictContainer';
import {firstGradeWinter} from './BoxChart.fixture.js';
import CohortChart from './CohortChart';

beforeEach(mockFetch);

export function mockFetch() {
  fetchMock.restore();
  fetchMock.get('express:/api/students/:student_id/reader_profile_cohort_json', {
    cells: {
      '2018-winter': { value: 101, stats: { p: 20, n_lower: 1, n_equal: 0, n_higher: 4 } },
      '2018-spring': { value: 101, stats: { p: 40, n_lower: 2, n_equal: 0, n_higher: 3 } },
      '2019-fall':   { value: 132, stats: { p: 60, n_lower: 2, n_equal: 2, n_higher: 1 } },
      '2019-winter': { value: 132, stats: { p: 40, n_lower: 2, n_equal: 0, n_higher: 3 } },
    }
  });
}

export function testProps(props = {}) {
  return {
    studentId: 42,
    gradeNow: '1',
    benchmarkAssessmentKey: 'f_and_p_english',
    readerJson: {
      access: {},
      services: [],
      iep_contents: null,
      feed_cards: [],
      "current_school_year": 2019,
      "benchmark_data_points": firstGradeWinter
    },
    ...props
  };
}

export function testRender(props = {}) {
  return withNowContext('2020-01-03T14:36:34.501Z',
    <PerDistrictContainer districtKey="somerville">
      <CohortChart {...props} />
    </PerDistrictContainer>
  );
}

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(testRender(testProps()), el);
});


it('snapshots', done => {
  expectSnapshotToMatchAfterTick(testRender(testProps()), done);
});
