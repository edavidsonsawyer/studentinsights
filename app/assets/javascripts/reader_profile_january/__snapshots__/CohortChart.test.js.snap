// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
<div
  className="CohortChart"
>
  <div>
    <div
      className="BoxChart"
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "fontSize": 12,
        }
      }
    >
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "cursor": "default",
                "display": "flex",
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
                "outline": "1px solid #ccc",
                "zIndex": 0,
              }
            }
            title={null}
          />
          <div
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "cursor": "default",
                "display": "flex",
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
                "outline": "1px solid #ccc",
                "zIndex": 1,
              }
            }
            title="20th"
          >
            20th
          </div>
          <div
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "cursor": "default",
                "display": "flex",
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
                "outline": "1px solid #ccc",
                "zIndex": 1,
              }
            }
            title="35th"
          >
            35th
          </div>
        </div>
        <div
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          Kindergarten
          , 
          2018
        </div>
      </div>
      <div
        style={
          Object {
            "flex": 1,
            "paddingLeft": 20,
          }
        }
      >
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
            }
          }
        >
          <div
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "cursor": "default",
                "display": "flex",
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
                "outline": "1px solid #ccc",
                "zIndex": 1,
              }
            }
            title="45th"
          >
            45th
          </div>
          <div
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "cursor": "default",
                "display": "flex",
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
                "outline": "1px solid #ccc",
                "zIndex": 1,
              }
            }
            title="41st"
          >
            41st
          </div>
          <div
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f8f8f8",
                "cursor": "default",
                "display": "flex",
                "flex": 1,
                "height": 40,
                "justifyContent": "center",
                "outline": "1px solid #ccc",
                "zIndex": 0,
              }
            }
            title={null}
          />
        </div>
        <div
          style={
            Object {
              "marginTop": 5,
            }
          }
        >
          1st grade
          , 
          2019
        </div>
      </div>
    </div>
  </div>
</div>
`;
